packages/react/src/button/button.styles.ts

```diff
@@ -763,7 +763,106 @@ export const StyledButton = styled(
         css: {
           br: '$pill'
         }
-      }
+      },
+      //Variants Interaction with Disabled State
+      {
+        color: 'success',
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4'
+        }
+      },
+      {
+        color: 'primary',
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4'
+        }
+      },
+      {
+        color: 'secondary',
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4'
+        }
+      },
+      {
+        color: 'warning',
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4'
+        }
+      },
+      {
+        color: 'error',
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4'
+        }
+      },
+      {
+        color: 'gradient',
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4'
+        }
+      },
+      {
+        bordered: true,
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4',
+          borderStyle: 'none'
+        }
+      },
+      {
+        flat: true,
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4'
+        }
+      },
+      {
+        color: 'light',
+        disabled: true,
+        css :{
+          bg: '$accents2',
+          color: '$accents4'
+        }
+      },
+      {
+        rounded: true,
```

Comment:
Remove this compound variant because `rounded` variant doesn't affect the colours